conn.go

```diff
@@ -1458,6 +1458,10 @@ func (c *Conn) execEx(ctx context.Context, sql string, options *QueryExOptions,
 			return "", err
 		}
 	} else if options != nil && len(options.ParameterOIDs) > 0 {
+		if err := c.ensureConnectionReadyForQuery(); err != nil {
+			return "", err
+		}
```

Comment:
This block could go near the top of either `execEx` or `ExecEx` instead (similar to `QueryEx` or `prepareEx`, but I believe this is the only case where `ensureConnectionReadyForQuery` isn't always called in some other place before writing to the connection.